### PR TITLE
allow to add timestamp suffix to log path

### PR DESCRIPTION
**Pull request by gfreezy**

### CODE REVIEW

**Comment by BourgondAries** [comment]
Not sure why this is needed. File metadata already contains the timestamp.

**Comment by gfreezy** [comment]
After restarting the process, the new log starts with `xxxx.log` which will overwrite the previous run's log. If we want to keep the previous run's log, we have to rename `xxxx.log` to `xxx.log.1`, then `xxx.log.1` will be overwritten. So we have to rename all the `xxx.log.xx` which is very verbose.

With timestamp suffix, renaming the previous run's `xxx.log` to `xxx.log.timestamp-suffix` is enough.

**Comment by BourgondAries** [comment]
That's a good case, thanks for explaining. Can you please resolve the conflicts currently?

**Comment by Ploppz** [comment]
@gfreezy  I think I'll build on this branch in an upcoming PR. Will you fix the conflicts? Otherwise I'll just include your changes in my PR.

**Comment by gfreezy** [comment]
@BourgondAries Conflicts resolved.

**Comment by Ploppz** [comment]
One thought:

How does this work if you use it in an application that might stop and then start again?
It will forget `log_paths` and thus fail to remove old files. Maybe we need to do a directory listing in `new()` and parse file names to construct `log_paths`? What do you think?

File: README.md [hunk 1]
```diff
@@ -9,3 +63,4 @@ This project is licensed under the [MIT license].
 Unless you explicitly state otherwise, any contribution intentionally submitted
 for inclusion in file-rotate by you, shall be licensed as MIT, without any additional
 terms or conditions.
+
```
Comment (BourgondAries) [inline README.md:66]: Trailing newline

File: README.md [hunk 2]
```diff
@@ -3,13 +3,13 @@
 Rotate files with timestamp postfix.
 
 ## Rotating by Lines #
-  
+
```
Comment (BourgondAries) [inline README.md:6]: What's going on here? Replacing \n by \n\r? We should not have this in the diff.

File: Cargo.toml [hunk 3]
```diff
@@ -9,6 +9,9 @@ repository = "https://github.com/BourgondAries/file-rotate"
 keywords= ["log", "rotate", "logrotate"]
 license = "MIT"
 
+[dependencies]
+chrono = "0.4.11"
```
Comment (BourgondAries) [inline Cargo.toml:13]: Put this behind a feature.

**Review by BourgondAries — CHANGES_REQUESTED** [review]
If you'd like this added please put it behind a feature to keep the core as simple as possible. I'd like to keep it backwards compatible too.
Please squash the commits.

**Comment by Ploppz** [comment]
Will be difficult in this PR to put chrono behind a feature, because there is no option anymore to use a simple count as suffix. I'm fixing that in my branch.

**Comment by gfreezy** [comment]
It's difficult to be backwards compatible. I will close the PR.  It is a lot of work to implement features like https://docs.rs/log4rs/1.0.0/log4rs/append/rolling_file/struct.RollingFileAppenderDeserializer.html#configuration or https://docs.rs/tracing-appender/0.1.2/tracing_appender/ .

**Pull request closed without merging**